python_modules/libraries/dagster-duckdb/dagster_duckdb/io_manager.py

```diff
@@ -279,16 +279,23 @@ def get_select_statement(table_slice: TableSlice) -> str:
     @staticmethod
     @contextmanager
     def connect(context, _):
+        config = context.resource_config["connection_config"]
+
+        # support for `custom_user_agent` was added in v1.0.0
+        # https://github.com/duckdb/duckdb/commit/0c66b6007b736ed2197bca54d20c9ad9a5eeef46
+        if duckdb.__version__ > "1.0.0":
+            config = {
+                "custom_user_agent": "dagster",
+                **config,
```

Comment:
is the intention that if a user specifies their own `custom_user_agent` that it overrides us setting it as `dagster`?

Reply:
yes, that's right.